mifosng-android/src/main/java/com/mifos/mifosxdroid/navigation/PasscodeNavGraph.kt

```diff
@@ -23,20 +26,33 @@ internal fun NavGraphBuilder.passcodeNavGraph(navController: NavHostController)
         passcodeRoute(
             onForgotButton = {
                 navController.popBackStack()
-                navController.navigate(MifosNavGraph.MAIN_GRAPH)
+                navController.navigateToMainGraph()
             },
             onSkipButton = {
                 navController.popBackStack()
-                navController.navigate(MifosNavGraph.MAIN_GRAPH)
+                navController.navigateToMainGraph()
             },
             onPasscodeConfirm = {
                 navController.popBackStack()
-                navController.navigate(MifosNavGraph.MAIN_GRAPH)
+                navController.navigateToMainGraph()
             },
             onPasscodeRejected = {
                 navController.popBackStack()
-                navController.navigate(MifosNavGraph.MAIN_GRAPH)
+                navController.navigateToMainGraph()
             },
         )
     }
 }
+fun NavController.navigateToMainGraph(){
+    val options = navOptions {
+        popUpTo(graph.findStartDestination().id) {
+            saveState = false
+        }
+        launchSingleTop = true
+        restoreState = false
+    }
+    navigate(MifosNavGraph.MAIN_GRAPH , options)
+
+
+
+}
```

Comment:
we don't need it. 
The error can be fixed by changing the Navroute of `MifosNavGraph.MainGraph`.

Reply:
Done Bhai